src/App.js

```diff
@@ -1,26 +1,97 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from 'react'
+import creatureData from './data.js'
+import ImageItem from './ImageItem.js'
+import Header from './Header.js'
 
-function App() {
+
+
+export default class App extends Component {
+  state = { selected: null };
+  
+  handleChange = (e) => {
```

Comment:
"Lastly, in order to get the Horns filter, I basically copy and pasted the other filter. I'm sure there is a way to make it less repetitive, but whenever I tried to bundle two .map statements, it broke."

You can combine your filters, but you need an event-handler for each filter input. Currently, you're tracking a change to either filter input with the same event handler.

So, in the onChange on line 45 below, you would have something like onChange={this.handleHornChange}.
Here, you would add a handleHornChange function...
handleHornChange = (e) => {
    this.setState({
      horns: e.target.value
    });
  }

THEN you can chain your filters together...
const filteredCreatures = creatureData
    .filter(creature => creature.horns >= this.state.horns)
    .filter(creature => creature.description.includes(this.state.description))

THEN you return your imageList component with the filtered creatures...
<ImageList creatures={filteredCreatures}/>